tests/phpunit/StorageExtractorTest.php

```diff
@@ -0,0 +1,326 @@
+
+        $configFile = [
+            'action' => 'run',
+            'parameters' => [
+                '#token' => getenv('KBC_TEST_TOKEN'),
+                'url' => getenv('KBC_TEST_URL'),
+            ],
+        ];
+        $baseDir = $temp->getTmpFolder();
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        $app->run();
+        self::assertFileExists($baseDir . '/out/tables/some-table-1.csv');
+        $csv = new CsvFile($baseDir . '/out/tables/some-table-1.csv');
+        $rows = iterator_to_array($csv);
+        sort($rows);
+        self::assertEquals(
+            [
+                ['1', 'a'],
+                ['2', 'b'],
+                ['3', 'c'],
+                ['id', 'name'],
+            ],
+            $rows
+        );
+        self::assertFileExists($baseDir . '/out/tables/some-table-1.csv.manifest');
+        $data = json_decode(file_get_contents($baseDir . '/out/tables/some-table-1.csv.manifest'), true);
+        self::assertEquals([], $data);
+    }
+
+    public function testInvalidToken(): void
+    {
+        $temp = new Temp('ex-storage');
+        $temp->initRunFolder();
+        $fs = new Filesystem();
+        $configFile = [
+            'action' => 'run',
+            'parameters' => [
+                '#token' => 'invalid',
+                'url' => getenv('KBC_TEST_URL'),
+            ],
+        ];
+        $baseDir = $temp->getTmpFolder();
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        self::expectException(UserException::class);
+        self::expectExceptionMessage('Invalid access token');
+        $app->run();
+    }
+
+    public function testInvalidTokenPermissions(): void
+    {
+        $temp = new Temp('ex-storage');
+        $temp->initRunFolder();
+        $fs = new Filesystem();
+        $configFile = [
+            'action' => 'run',
+            'parameters' => [
+                '#token' => getenv('KBC_TEST_WRITE_TOKEN'),
+                'url' => getenv('KBC_TEST_URL'),
+            ],
+        ];
+        $baseDir = $temp->getTmpFolder();
+        $fs->dumpFile($baseDir . '/config.json', \GuzzleHttp\json_encode($configFile));
+        putenv('KBC_DATADIR=' . $baseDir);
+        $app = new Component(new NullLogger());
+        self::expectException(UserException::class);
+        self::expectExceptionMessage('The token does not have only read permissions to the bucket in.c-ex-storage');
+        $app->run();
+    }
+
+    public function testTableFilter(): void
+    {
+        $temp = new Temp('ex-storage');
+        $temp->initRunFolder();
+        $fs = new Filesystem();
+        $fs->dumpFile($temp->getTmpFolder() . '/tmp.csv', "\"id\",\"foo\"\n\"1\",\"aa\"\n\"2\",\"bb\"\n\"3\",\"cc\"\n");
```

Comment:
Jsem jediný, kdo má z těch backslashů brainfuck? Co 

```php
implode("\n", [
'"id", "foo"',
'"1", "a"',
// ...
];
```
Nebo `$fs->copy($temp->getTmpFolder() . '/tmp.csv', __DIR__ . '/fixtures/1.csv')`

Reply:
me to tady prijde v pohode, protoze obsah toho souboru je uplne bezvyznamnej, idealni by bylo mit  v phpunit metodu `assertStorageTableSame`